couchbase/admin/query/v1/query.proto

```diff
@@ -0,0 +1,111 @@
+syntax = "proto3";
+
+option go_package = "github.com/couchbase/goprotostellar/genproto/admin_query_v1;admin_query_v1";
+option ruby_package = "Couchbase::Protostellar::Generated::Admin::Query::V1";
+option php_namespace = "Couchbase\\Protostellar\\Generated\\Admin\\Query\\V1";
+option java_package = "com.couchbase.client.protostellar.admin.query.v1";
+option java_multiple_files = true;
+
+package couchbase.admin.query.v1;
+
+service QueryAdminService {
+  rpc GetAllIndexes(GetAllIndexesRequest) returns (GetAllIndexesResponse) {}
+
+  rpc CreatePrimaryIndex(CreatePrimaryIndexRequest)
+      returns (CreatePrimaryIndexResponse) {}
+  rpc CreateIndex(CreateIndexRequest) returns (CreateIndexResponse) {}
+
+  rpc DropPrimaryIndex(DropPrimaryIndexRequest)
+      returns (DropPrimaryIndexResponse) {}
+  rpc DropIndex(DropIndexRequest) returns (DropIndexResponse) {}
+
+  rpc BuildDeferredIndexes(BuildDeferredIndexesRequest)
+      returns (BuildDeferredIndexesResponse) {}
+}
+
+enum IndexType {
+  INDEX_TYPE_VIEW = 0;
+  INDEX_TYPE_GSI = 1;
+}
+
+enum IndexState {
+  INDEX_STATE_DEFERRED = 0;
+  INDEX_STATE_BUILDING = 1;
+  INDEX_STATE_PENDING = 2;
+  INDEX_STATE_ONLINE = 3;
+  INDEX_STATE_OFFLINE = 4;
+  INDEX_STATE_ABRIDGED = 5;
+  INDEX_STATE_SCHEDULED = 6;
+}
+
+message GetAllIndexesRequest {
+  optional string bucket_name = 1;
+  optional string scope_name = 2;
+  optional string collection_name = 3;
+}
+
+message GetAllIndexesResponse {
+  message Index {
+    string bucket_name = 1;
+    string scope_name = 2;
+    string collection_name = 3;
```

Comment:
optional for fields 2 and 3?  Or will they return "_default"?

Reply:
I originally had these optional, but then afterwards realized that even though at index creation time they are optional, once the index is created, it does explicitly pick a collection for the index (being _default/_default if you didn't specify the option at creation time).

Reply:
Yep, makes sense

Reply:
Going to file this under the caveats that will affect https://couchbasecloud.atlassian.net/browse/ING-517.